### PR TITLE
Wait for testhost stderr to be available if connection is broken. (#772)

### PR DESCRIPTION
**Pull request by sbaid**
This allows to provide user information if the test host has crashed.

Don't send stderr to console runner output if it's null or empty.

### CODE REVIEW

**Comment by codito** [comment]
@dotnet-bot test Ubuntu16.04 please

**Comment by codito** [comment]
Looking into the build failure.

File: src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs [hunk 1]
```diff
@@ -396,10 +396,34 @@ private string TryReceiveRawMessage()
...
+                }
+                else
+                {
+                    // Test host process has not exited yet. We will wait for exit to allow us gather
+                    // standard error
+                    var processWaitEvent = new ManualResetEventSlim();
+                    try
+                    {
+                        EqtTrace.Info("TestRequestSender: Waiting for test host to exit.");
+                        processWaitEvent.Wait(TimeSpan.FromSeconds(10), this.clientExitCancellationSource.Token);
```
Comment (Faizan2304) [inline src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs]: If in 10 sec testhost doesn't exit, what will happen to testhost process? Shouldn't we kill testhost explicitly?

Comment (codito) [inline src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs]: Yes, we should. Will bring this in a future PR. cc: @smadala

**Pull request merged**